src/odemis/gui/cont/streams.py

```diff
@@ -2678,39 +2719,39 @@ def addSpectrum(self, name=None, detector=None):
         logging.debug("Adding spectrum stream for %s", detector.name)
 
         spg = self._getAffectingSpectrograph(detector)
+
+        # band should be set to the pass-through by default
+        axes = {"wavelength": ("wavelength", spg),
+                "grating": ("grating", spg),
+                "slit-in": ("slit-in", spg),
+               }
+
```

Comment:
You need to drop these axes if there are not available on the hardware, here, before creating the stream. 
Maybe you could put this functionality into a common method named like `._filter_axes(axes)`. It would take the `axes` parameter passed to the stream, and return the same dict, but without the invalid entries (ie, component is None or component doesn't have this axis).

Reply:
You mean universal for all of the times you add axes?

Reply:
All the time that the axes are added in the SPARCStreamController.